src/wasm-ast-parser.y

```diff
@@ -1242,21 +1249,6 @@ raw_module :
           }
         }
       }
-
-      for (i = 0; i < $4->imports.size; ++i) {
-        WasmImport* import = $4->imports.data[i];
-        if (import->kind == WASM_EXTERNAL_KIND_FUNC &&
-            wasm_decl_has_func_type(&import->func.decl) &&
-            is_empty_signature(&import->func.decl.sig)) {
-          WasmFuncType* func_type =
-              wasm_get_func_type_by_var($4, &import->func.decl.type_var);
-          if (func_type) {
-            import->func.decl.sig = func_type->sig;
-            import->func.decl.flags |=
-                WASM_FUNC_DECLARATION_FLAG_SHARED_SIGNATURE;
-          }
-        }
-      }
```

Comment:
This is pretty subtle; the functions in the import array are actually already included in the funcs array, so this behavior is actually duplicated.